controllers/users.js

```diff
@@ -1,53 +1,56 @@
 const bcrypt = require('bcryptjs');
 const jwt = require('jsonwebtoken');
 
+const NotFoundError = require('../errors/error_not_found');
+const Error500 = require('../errors/error_500');
+const Error401 = require('../errors/error_Auth');
 const User = require('../models/user');
-/*
+
 const { JWT_SECRET } = require('../secret.js');
-const token = jwt.sign({ _id: user._id }, JWT_SECRET, { expiresIn: '7d' });
-*/
+
 // отправим токен, браузер сохранит его в куках
 
 module.exports.getAllUsers = (req, res) => {
   User.find({})
     .then((user) => {
       if (user.length === 0) {
-        return res.status(404).send({ message: 'База данных user пуста! ' });
+        return (new NotFoundError({ message: 'База данных user пуста! ' }));
       }
       return res.send({ data: user });
     })
-    .catch((error) => res.status(500).send({ message: error.message }));
+    .catch((new Error500('На сервере произошла ошибка')));
 };
 
 module.exports.createUser = (req, res) => {
-  const { name, about, avatar, email, password } = req.body;
+  const {
+    name, about, avatar, email, password,
+  } = req.body;
   if (password.length > 11) {
     bcrypt.hash(password, 10)
-      .then((hash) => User.create({name, about, avatar, email, password: hash}))
-      .then(({name, about, avatar, email}) => res.send({name, about, avatar, email }))
-      .catch(() => res.status(404).send({ message: 'Не удалось создать пользователя' }));
-  } else {
-    res.status(500).send({ message: 'Слишком короткий пароль!' });
+      .then((hash) => User.create({
+        name, about, avatar, email, password: hash,
+      }))
+      .then((user) => res.send({ data: user.omitPrivate() }))
+      .catch(new Error500('Не удалось создать пользователя'));
   }
 };
 
 module.exports.getUser = (req, res) => {
   User.findById(req.params.userId)
     .then((userId) => {
       if (!userId) {
-        res.status(404).send({ message: 'Такого пользователя нет' });
+        throw new NotFoundError({ message: 'Такого пользователя нет' });
       } else {
         res.send({ userId });
       }
     })
-    .catch(() => res.status(500).send({ message: 'Нет пользователя с таким id' }));
+    .catch(new Error500('Нет пользователя с таким id'));
```

Comment:
Нужно исправить: ошибка не обрабатывается централизованным обработчиком.
Централизованная обработка ошибок запустится только если будет вызван `next`:
```suggestion
    .catch(next);
```

Только нужно не забыть объявить `next` здесь:
https://github.com/MaximMaslovVik/15_sprint/blob/2ad864e57b00f8564f3658374d392b861c3b10fb/controllers/users.js#L38